File: crawler-worker/src/document-number.ts
export function extractDocumentNumber(text: string): string | null {
  const normalized = text.normalize('NFKC').replace(/[【\[]/g, '〔').replace(/[】\]]/g, '〕').replace(/\s+/g, '');
  const match = normalized.match(/([\u4e00-\u9fffA-Za-z]{1,24})〔(19\d{2}|20\d{2})〕(\d{1,6})号/);
  if (!match) return null;
  const prefix = match[1].replace(/^(关于印发|关于发布|关于|印发|发布)/, '');
  return `${prefix}〔${match[2]}〕${match[3]}号`;
}

File: scrapling-worker/app/document_number.py
from __future__ import annotations

import re
import unicodedata


_PATTERN = re.compile(r"([\u4e00-\u9fffA-Za-z]{1,24})〔(19\d{2}|20\d{2})〕(\d{1,6})号")
_PREFIX_NOISE = re.compile(r"^(关于印发|关于发布|关于|印发|发布)")


def extract_document_number(text: str) -> str | None:
    normalized = unicodedata.normalize("NFKC", text)
    normalized = normalized.replace("【", "〔").replace("[", "〔").replace("】", "〕").replace("]", "〕")
    normalized = re.sub(r"\s+", "", normalized)
    match = _PATTERN.search(normalized)
    if not match:
        return None
    prefix, year, number = match.groups()
    prefix = _PREFIX_NOISE.sub("", prefix)
    return f"{prefix}〔{year}〕{number}号"
